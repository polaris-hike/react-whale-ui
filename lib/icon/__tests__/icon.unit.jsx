import renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../icon';
import { mount } from 'enzyme';

describe('Icon', () => {
  it('是个 svg', () => {
    const json = renderer.create(<Icon name="wechat"/>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('onclick', () => {
    const fn = jest.fn();
    const wrapper = mount((
      <Icon onClick={fn} />
    ));
    //const component = mount(<Icon name="wechat" onClick={fn}/>);
    wrapper.find('svg').simulate('click');
    expect(fn).toBeCalled();
  })
});