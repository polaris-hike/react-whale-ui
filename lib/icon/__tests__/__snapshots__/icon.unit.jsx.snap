// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 是个 svg 1`] = `
<svg
  aria-hidden="true"
  className="whale-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
